### PR TITLE
Add config options to manage variables expansion

### PR DESCRIPTION
**Pull request by irizzant**
**What this PR does**:
This PR adds support for specifying config options to manage variables expansion in a single point.

For example, to enable S3 storage backend simply add the following to the `_config` object:
```
...
local envVar = k.core.v1.envVar,
...
variables_expansion: true,
variables_expansion_env_mixin: [
          envVar.withName('S3_ACCESS_KEY')
          + envVar.valueFrom.secretKeyRef.withKey('S3_ACCESS_KEY')
          + envVar.valueFrom.secretKeyRef.withName('minio-secret'),
          envVar.withName('S3_SECRET_KEY')
          + envVar.valueFrom.secretKeyRef.withKey('S3_SECRET_KEY')
          + envVar.valueFrom.secretKeyRef.withName('minio-secret'),
        ],
```
where `S3_ACCESS_KEY` and `S3_SECRET_KEY` are taken for a secret named `minio-secret`.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [X] Tests updated
- [X] Documentation added
- [X] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ## main / unreleased
 
+* [ENHANCEMENT]: Improve variables expansion support [#1212](https://github.com/grafana/tempo/pull/1212) @irizzant
```
Comment (joe-elliott) [inline CHANGELOG.md:3]: Please move this to the enhancement section of the changelog.

**Review by joe-elliott — COMMENTED** [review]
nice addition. just asking for some minor cleanup to the changelog.

File: CHANGELOG.md (hunk 1 above)
Comment (irizzant) [reply CHANGELOG.md]: @joe-elliott done as requested

**Comment by joe-elliott** [comment]
@irizzant CI was broken in main. If you could rebase/merge main CI should pass and we can get this merged.

**Comment by irizzant** [comment]
@joe-elliott that makes sense...I updated the branch as you suggested it should be ok then

**Review by joe-elliott — APPROVED** [review]
Nice. Thanks for the contribution!

**Pull request merged**